src/nunit.analyzers/Extensions/AttributeArgumentSyntaxExtensions.cs

```diff
@@ -64,6 +64,11 @@ internal static bool CanAssignTo(this AttributeArgumentSyntax @this, ITypeSymbol
                     {
                         canConvert = TimeSpan.TryParse(argumentValue as string, out _);
                     }
+                    else if (argumentType.SpecialType == SpecialType.System_String &&
+                        model.Compilation.GetTypeByMetadataName(typeof(DateTimeOffset).FullName).IsAssignableFrom(targetType))
+                    {
+                        canConvert = DateTimeOffset.TryParse(argumentValue as string, out _);
```

Comment:
Note to reviewer: I used `TryParse` here, but I think the underlying NUnit implementation uses `TypeConverter.CanConvertFrom`. Please let me know if you'd prefer I update it.

Reference code: https://github.com/nunit/nunit/blob/master/src/NUnitFramework/framework/Internal/ParamAttributeTypeConversions.cs#L138-L143

Reply:
Actually, I don't know the exact differences between `DateTimeOffset.TryParse` and `TypeConverter.CanConvertFrom`, so I'm unsure about this. But I think it will be the safest to have the same implementation, if it is not too cumbersome. Also I think we should use `CultureInfo.InvariantCulture`.

Reply:
Thanks for the feedback. I think I have this working now using `TypeConverter` + `CultureInfo.InvariantCulture` :)